cartographer_ros/cartographer_ros/rosbag_validate_main.cc

```diff
@@ -147,19 +216,65 @@ void Run(const std::string& bag_filename, const bool dump_timing) {
                            << message.getTime().toNSec() << "\t"
                            << time.toNSec() << std::endl;
     }
+
+    double duration_serialization_sensor = (time - message.getTime()).toSec();
+    if (std::abs(duration_serialization_sensor) > 0.1) {
+      std::stringstream stream;
+      stream << "frame_id \"" << frame_id << "\" on topic "
+             << message.getTopic() << " has serialization time "
+             << message.getTime() << " but sensor time " << time
+             << " differing by " << duration_serialization_sensor << " s.";
+      if (std::abs(duration_serialization_sensor) > 0.5) {
+        LOG_FIRST_N(ERROR, 3) << stream.str();
+      } else {
+        LOG_FIRST_N(WARNING, 1) << stream.str();
+      }
+    }
   }
   bag.close();
 
+  if (num_imu_messages > 0) {
+    double average_imu_acceleration = sum_imu_acceleration / num_imu_messages;
+    if (std::isnan(average_imu_acceleration) ||
+        average_imu_acceleration < 9.5 || average_imu_acceleration > 10.5) {
```

Comment:
Here and below should we collect these number constants in one place as constants to make it more convenient to change them?

Reply:
Done